utils/sl/setup_local.sh

```diff
@@ -105,6 +101,7 @@ for item in $list; do
     fi
 done
 
+# shellcheck disable=SC2181
 in_list "${SL_PREFIX}/bin" "${old_path}"
 if [ $? -eq 0 ]; then
```

Comment:
What was wrong with your previously proposed
```suggestion
if ! in_list "${SL_PREFIX}/bin" "${old_path}"
```

This is the more idiomatic syntax.

Reply:
It wasn't working, there was a unit test failure as a result of this change.

Reply:
It's odd that it wasn't working.  They are functionally the equivalent when `set -e` isn't being used (which is a bad thing in it's own right) and the existing syntax won't work if `set -e` is being used.

So `set -e` not being in use here is another issue.

Reply:
This script is intended to be sourced by the user to setup their env.  If you use set -e or exit then it logs the user out which is not the desired behaviour.

Reply:
Neither do I, and yet it was the only change in that build.  Like I say my goal is not to fix or work on the existing warnings but improve our processes so that we don't end up with more in the codebase.  It would be well worth a ticket to resolve all currently suppressed shellcheck warnings.